src/status.rs

```diff
@@ -103,13 +104,128 @@ fn estimated_total_task_time(
 	}
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum ProgressFormat {
+	None,
+	Text,
+	ASCIIBar,
+	HighResBar,
+}
+
+#[derive(Debug)]
+pub struct ParseProgressFormatError {
+	value: String,
+}
+
+impl core::str::FromStr for ProgressFormat {
+	type Err = ParseProgressFormatError;
+	fn from_str(s: &str) -> Result<Self, ParseProgressFormatError> {
+		let s = s.to_lowercase();
+		if s == "none" {
```

Comment:
Instead of a series of `if`s, just use a match:

```rust
match s.to_lowercase() {
    "none" => Ok(ProgressFormat::None),
    "text => Ok(ProgressFormat::Text),
    ...
    value => Err(ParseProgressFormatError { value }),
}
```

Reply:
Oh, I didn't know you could do that!